site/app/libraries/database/DatabaseQueries.php

```diff
@@ -1594,22 +1590,42 @@ public function getAllElectronicGradeablesIds() {
     // }
 
     /**
-     * Create a new team id and team in gradeable_teams for given gradeable, add $user_id as a member
-     * @param string $g_id
+     * Generates a new team id from a user's name
      * @param string $user_id
-     * @param integer $registration_section
-     * @param integer $rotating_section
-     * @return string $team_id
+     * @return string
      */
-    public function createTeam($g_id, $user_id, $registration_section, $rotating_section) {
+    public function generateTeamId(string $user_id) {
```

Comment:
Sort of minor, but generating team ids and creating them in another function opens you to race conditions... would highly recommend combining the functions

Reply:
Or at least don't make this one public

Reply:
that's true; i'll work on it.  Its part of handling the issue of 'something threw an exception, but the teams were created already.'